### ISSUE TITLE
Feature Request: Native Mode Samba user changeable

### ISSUE THREAD

**Issue by cpp339** [jellyfin/jellyfin-kodi#182]
It would be great when the Samba username/password for the native mode could be set by the user. The usage of the the guest-user without a password is in some environments not possible.

**Comment by shemanaev** [comment]
Is there any reason why credentials specified when [adding remote source](https://kodi.wiki/view/Adding_video_sources#Adding_Remote_sources) wouldn't work? Despite that being said in instruction

**Comment by cpp339** [comment]
if the user only has to add the remote source with credentials and native mode will work without any additional configuration, I will be happy too.

**Comment by Shadowghost** [comment]
I'd also be interested in this. Either way it would be nice to have (I don't want to open up my network storage to random people in my network by enableing guest access).

**Comment by netpok** [comment]
One can pass the authentication data to the client by specifying the remote mapping like:
`\\user:password@server\share\location`

This is better than using a guest user but an option to use the already entered username and password would be nice.

**Comment by Shadowghost** [comment]
Thanks for the info. I thought that would not work since the documentation explicitly states to use the guest user.

**Comment by mcarlton00** [comment]
As a disclaimer: none of the current devs actually use Native mode right now, so our knowledge of it's inner workings is a bit thin.  We simply don't have the hardware or a need for it in our local setups.

The documentation for native mode was written by another generous contributor and they likely only had a need for guest mode and no authentication, which is why it was specified.  If it's as easy as just passing in `user:password@` when mounting the volume then that's perfect and can easily be amended in the docs over [here](https://github.com/jellyfin/jellyfin-docs).

@netpok By "already entered username and password", I'm assuming you mean the same login credentials as your Jellyfin user, which I hesitate to do since (in my opinion) that would lead users to reusing passwords for multiple things, and I hate encouraging bad behavior.  I'm open to having my mind changed on this if there's a good argument for having it share credentials.

**Comment by netpok** [comment]
@mcarlton00 While I'm not using it because I'm too lazy to configure it, the use case would be a system where both the samba and jellyfin are linked to the same authentication provider (LDAP). So there wouldn't be any password reusing.

This was just an idea.

**Comment by mcarlton00** [comment]
That's a fair point, especially since we have the ldap auth plugin in the server, and hopefully others coming eventually. Thinking more on it, I do see an issue in sharing the credentials, though. You're adding the library to Kodi itself, not the add-on, so I suspect we won't be able to integrate that tightly and share the credentials. It's likely that we'll be limited to the user: password syntax, but we won't know for sure until somebody gets around to doing some digging for it

**Comment by netpok** [comment]
I sadly don't have the time right now to look into this, but from a security standpoint it would be the best to use some kind of app password system so an app has its password which can be revoked/changed without affecting anything else (maybe this is already possible with the api keys, haven't checked). But this would also mean that my previous idea would not work.

**Comment by cpp339** [comment]
> One can pass the authentication data to the client by specifying the remote mapping like:
> `\\user:password@server\share\location`

Thanks for the hint. The only issue is that you can see the passphrase within the settings of the remote mapping.

After this it looks like it's not necessary to add the samba shares to kodi as described in the documentation:

> 3. Add in your SMB/Samba shares into Kodi like you normally would when not using the Jellyfin plugin, setting the content type, but it is not necessary to import that data since this will be handled by the Jellyfin plugin.

**Comment by mcarlton00** [comment]
So just to recap, it seems like there's two options here:

* In the Jellyfin server, specify your remote path as `\\user:password@server\Share`
* In Kodi, set up a remote mount with `Settings -> File Manager -> Add Source -> Browse -> Add network location` where it will ask for a username/password there.

I've just tested both methods and they both work fine.  I'm inclined to think the second one is better since you won't be passing network credentials in plain text, but it will take some effort to make it more user friendly.

Every Kodi window has an ID associated to it, so in theory it should be possible to have the JF for Kodi open the new source dialog and prompt users there for login.  It may also be possible to pull the remote path substitution from the server and automatically add a local source, but I'm not seeing an option for that in Kodi's [list of commands](https://kodi.wiki/view/List_of_built-in_functions#System_built-in.27s) we have available through the python interface.

TLDR: Maybe possible, but more thought is needed

**Comment by oddstr13** [comment]
Native mode takes the free-form text field provided (per library directory) by the Jellyfin server, and joins it with the relative file path of the media.
This means you can point this at any base path that Kodi supports, including (but not limited to):
 - Windows shares (SMB)
   `\\jellyfin\Movies`
 - locally mounted path (anything mountable by Linux)
   `/mnt/jellyfin/Movies`
 - http(s)
   `http://jellyfin/Movies`
  
I am for this reason closing this feature request as out of scope.
